src/webhook_command_utils.rs

```diff
@@ -0,0 +1,77 @@
+use crate::webhook::Webhook;
+
+#[derive(PartialEq, Debug)]
+pub struct BotCommand {
+    pub command: String,
+    pub args: String,
+    pub bot: String,
+}
+
+pub fn extract_commands(webhook: &Webhook, bots: &Vec<String>) -> Vec<BotCommand> {
+    // prevent bots from triggering commands potentially creating loops
+    if bots.contains(&webhook.author) {
+        println!("{} cannot trigger commands", webhook.author);
+        return vec![];
+    }
+
+    // only continue if comment was not deleted
+    if webhook.action.eq("deleted") {
+        println!("Exiting: comment deleted");
+        return vec![];
+    }
+
+    // ensure author of command has sufficient rights
+    let rights = &webhook.author_association;
+    if !(rights.eq("OWNER") || rights.eq("COLLABORATOR")) {
+        println!("permission denied: {}", rights);
+        return vec![];
+    }
+
+    parse_commands(&webhook.comment, bots)
+}
+
+fn parse_commands(text: &str, bots: &Vec<String>) -> Vec<BotCommand> {
+    let mut bot_commands: Vec<BotCommand> = Vec::new();
+    for bot in bots {
+        let atbot = &format!("@{bot} ");
+        text.lines()
+            .flat_map(|line| line.strip_prefix(atbot))
+            .flat_map(|line| line.split_once(' '))
```

Comment:
I think this misses the case of a command without arguments (should also be added to the tests).

Reply:
Yes i caught that in CI as well